File: js/mumbling.js
const accum = s => {
  let results = [];
  let stringStorage = [];
  let finalResults = [];
  let realFinalResults = [];
  const string = s.toLowerCase();

  // create an array with first letter upper followed by lower * i
  for (let i = 0; i < string.length; i++) {
    stringStorage.push(string[i]);
    if (i <= stringStorage.length) {
      results.push(string[i].repeat(1).toUpperCase());
      results.push(string[i].repeat(i));
    }
  }
  // join upper with lowercase mumble and put in main array
  for (let i = 0; i < results.length; i++) {
    let every2 = [];
    every2.push(results[i], results[i + 1]);
    finalResults.push(every2);
  }
  // join individual upper and lower
  for (let i = 0; i < results.length; i += 2) {
    realFinalResults.push(finalResults[i].join().replace(/,/g, ""));
  }
  // join entire array with hyphens
  return realFinalResults.join().replace(/,/g, "-");
};

File: python/mumbling.py
def accum(s):
    results = []
    for x in range(0, len(s)):
        # if first char is upper append, else make it upper and append
        results.append(s[x]) if s[x] == s[x].upper else results.append(s[x].upper())
        # multiply char by index and append
        results.append(x * s[x].lower()) if (x >= 1) else results
        # seperate values w/ a dash
        results.append("-") if (x != len(s)-1) else results
    # turn array into string w/ no spaces between elements
    return "".join(results)
